src/main/java/org/opensearch/knn/plugin/transport/UpdateBlockedModelTransportAction.java

```diff
@@ -0,0 +1,158 @@
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.TransportService;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+import lombok.AllArgsConstructor;
+
+import static org.opensearch.knn.common.KNNConstants.PLUGIN_NAME;
+
+/**
+ * Transport action used to update blocked modelIds (ModelGraveyard) on the cluster manager node.
+ */
+@Log4j2
+public class UpdateBlockedModelTransportAction extends TransportMasterNodeAction<UpdateBlockedModelRequest, AcknowledgedResponse> {
+    private UpdateBlockedModelExecutor updateBlockedModelExecutor;
+
+    @Inject
+    public UpdateBlockedModelTransportAction(
+        TransportService transportService,
+        ClusterService clusterService,
+        ThreadPool threadPool,
+        ActionFilters actionFilters,
+        IndexNameExpressionResolver indexNameExpressionResolver
+    ) {
+        super(
+            UpdateBlockedModelAction.NAME,
+            transportService,
+            clusterService,
+            threadPool,
+            actionFilters,
+            UpdateBlockedModelRequest::new,
+            indexNameExpressionResolver
+        );
+        this.updateBlockedModelExecutor = new UpdateBlockedModelExecutor();
+    }
+
+    @Override
+    protected String executor() {
+        return ThreadPool.Names.SAME;
+    }
+
+    @Override
+    protected AcknowledgedResponse read(StreamInput streamInput) throws IOException {
+        return new AcknowledgedResponse(streamInput);
+    }
+
+    @Override
+    protected void masterOperation(
+        UpdateBlockedModelRequest request,
+        ClusterState clusterState,
+        ActionListener<AcknowledgedResponse> actionListener
+    ) {
+        // ClusterManager updates blocked modelIds list based on request parameters
+        clusterService.submitStateUpdateTask(
+            PLUGIN_NAME,
+            new UpdateBlockedModelTask(request.getModelId(), request.isRemoveRequest()),
+            ClusterStateTaskConfig.build(Priority.NORMAL),
+            updateBlockedModelExecutor,
+            new ClusterStateTaskListener() {
+                @Override
+                public void onFailure(String s, Exception e) {
+                    actionListener.onFailure(e);
+                }
+
+                @Override
+                public void clusterStateProcessed(String source, ClusterState oldState, ClusterState newState) {
+                    actionListener.onResponse(new AcknowledgedResponse(true));
+                }
+            }
+        );
+    }
+
+    @Override
+    protected ClusterBlockException checkBlock(UpdateBlockedModelRequest request, ClusterState clusterState) {
+        return null;
+    }
+
+    /**
+     * UpdateBlockedModelTask is used to provide the executor with the information it needs to perform its task
+     */
+    @AllArgsConstructor
+    private static class UpdateBlockedModelTask {
```

Comment:
Seems this class has changed since I've reviewed the code, seems @Value fits better here as there are fields only and no other methods. Value includes getters for fields and also allArgsConstructor, please check doc on details https://projectlombok.org/features/Value

Reply:
ack